packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -219,13 +222,11 @@ export class RunningSummarizer implements IDisposable {
         // This will try to run lastSummary if needed.
         if (allowLastSummary && this.heuristicRunner?.shouldRunLastSummary()) {
             if (this.summarizingLock === undefined) {
-                // Note: trySummarizeOnce() expects to be called when lock is not taken.
-                this.generator.summarize(
+                this.trySummarizeOnce(
```

Comment:
With this change, there will be an extra call to checkSummarizeAgain() after this attempt. It should be fine because this.stopping is true. Should an explicit check for this.stopping be added in checkSummarizeAgain() to be sure?

Reply:
I'll refactor to check stopping first. Also will remove tryWhileSummarizing as it's not really needed - it should be totally safe to always run through heuristics - if previous summary was successful and not many ops happened since then, then it will be noop, otherwise it's find to run summaries back to back (weather we had concurrency or not).